pkg/rpc/server/listen.go

```diff
@@ -74,6 +95,10 @@ func ListenAndServe(c config.Config, fsCache cache.FSCache) error {
 	libHandler := rpcDetector.NewLibDetectorServer(initializeLibServer(), nil)
 	mux.Handle(rpcDetector.LibDetectorPathPrefix, withToken(withWaitGroup(libHandler), c.Token, c.TokenHeader))
 
+	// promHandler is for dealing with update the custom prometheus metrics
+	promHandler := promhttp.HandlerFor(MetricsRegistry, promhttp.HandlerOpts{})
```

Comment:
I think we should probably set a sane `timeout` as not setting one will keep connections open indefinitely. See here https://github.com/prometheus/client_golang/blob/master/prometheus/promhttp/http.go#L333-L342